src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.DecisionDagRewriter.cs

```diff
@@ -117,10 +117,8 @@ protected sealed class WhenClauseMightAssignPatternVariableWalker : BoundTreeWal
 
                 public bool MightAssignSomething(BoundExpression expr)
                 {
-                    if (expr == null || expr.ConstantValue != null)
-                    {
+                    if (expr == null)
```

Comment:
should we assert that we never reach here with a constant-valued expression?

Reply:
No. We will reach here with a constant-valued expression, and then line 124 `this.Visit(expr)` invokes the next method which handles it.

---
In reply to: [483127468](https://github.com/dotnet/roslyn/pull/47422#discussion_r483127468) [](ancestors = 483127468)